src/codegen/transformations.lisp

```diff
@@ -310,6 +310,43 @@
             (append funs binding-list-traversals)
             (list :bound-variables nil)))
 
+(defmethod tc:apply-substitution (subs (node node))
```

Comment:
Refactoring `tc:apply-substitution` like this breaks `(monomorphize)`, but I'm not sure why...
```
; caught COMMON-LISP:ERROR:
;   COMMON-LISP:READ error during COMMON-LISP:COMPILE-FILE:
;   
;     The value
;       #S(COALTON-IMPL/CODEGEN/AST:NODE-LITERAL :TYPE INTEGER :VALUE 2)
;     is not of type
;       COALTON-IMPL/CODEGEN/AST:NODE-VARIABLE
;   
;     (in form starting at line: 5, column: 0, position: 86)
```